src/h-adsorption-onto-pd.md

```diff
@@ -0,0 +1,2303 @@
+                        'label': 'Play',
+                        'method': 'animate'
+                    },
+                    {
+                        'args': [[None], {'frame': {'duration': 0, 'redraw': True}, 'mode': 'immediate', 'transition': {'duration': 0}}],
+                        'label': 'Pause',
+                        'method': 'animate'
+                    }
+                ],
+                'direction': 'left',
+                'pad': {'r': 10, 't': 87},
+                'showactive': False,
+                'type': 'buttons',
+                'x': 0.1,
+                'xanchor': 'right',
+                'y': 0,
+                'yanchor': 'top'
+            }
+        ],
+        title_text=f"Time: 0.00 ps",
+    )
+
+    # Add frames to the figure
+    fig.frames = frames
+
+    return fig
+
+```
+
+To test this out, we'll mock up a circular motion for the two molecules we visualsied earlier.
+
+```python
+# Constants
+num_timesteps = 1000
+
+# Initial conditions for two molecules
+initial_positions = [("H2", 1, 1, -1), ("H2", 2, 2, -1.5)]
+initial_bond_lengths = [angstrom_to_crystal(0.74, cell_parameters), angstrom_to_crystal(0.74, cell_parameters)]
+initial_theta = [np.pi / 2, np.pi / 4]  # Bond angles relative to surface normal
+initial_phi = [0, np.pi / 2]  # Orientation in the x-y plane
+
+# Storage for time-varying data
+h2_positions_over_time = []
+pd_positions_over_time = []
+h2_bond_lengths_over_time = []
+h2_theta_over_time = []
+h2_phi_over_time = []
+
+# Generate data for each times-tep
+for t in range(num_timesteps):
+    positions = []
+    bond_lengths = []
+    theta = []
+    phi = []
+
+    for i, pos in enumerate(initial_positions):
+        # Define the radius of the motion for each molecule
+        radius = 0.5 * (i + 1)  # Each molecule has a different radius
+    
+        # Calculate the center of the circle
+        center_x = pos[1]
+        center_y = pos[2] - radius  # Shift center to align initial position
+    
+        # Update position for circular motion
+        x = center_x + radius * np.sin(-2 * np.pi * t / num_timesteps)
+        y = center_y + radius * np.cos(-2 * np.pi * t / num_timesteps)
+        z = pos[3]  # z remains constant
+
+        positions.append(("H2", x, y, z))
+
+        # Bond length remains constant for simplicity
+        bond_lengths.append(initial_bond_lengths[i])
+
+        # Bond angles remain constant for simplicity
+        theta.append(initial_theta[i])
+        phi.append(initial_phi[i])
+
+    h2_positions_over_time.append(positions)
+    pd_positions_over_time.append(surface_pd_atoms)
+    h2_bond_lengths_over_time.append(bond_lengths)
+    h2_theta_over_time.append(theta)
+    h2_phi_over_time.append(phi)
+```
+
+We can now expand the molecule into it's atoms, create the bonds and combine the hydrogen and palladium together into a single array for plotting.
+
+```python
```

Comment:
I think it might be worthwhile to write a sentence or two about what the reader can expect to see here. Is this a kind of random path movement that the center of mass of each molecule undergoes? Sorry, it wasn't so clear to me when I looked at this. 

Also, it may be worthwhile to say something about the time here: this animation is run slower than the one below. Is there a reason? To see the dynamics better?

Reply:
On second look, here it looks like the molecules are moving not in straight paths -- but in the animation below they do seem to move in straight paths. Why is that?

Reply:
All good, there is a lot of code so the circular motion bit is easy to miss. I've reinforced it below:

>The previous block of code above simulates circular motion of the centre of mass of two molecules. In order to visualise the atoms and bonds, we have to expand the molecule into its atoms, create the bonds and combine the hydrogen and palladium together into a single array for plotting.

I've also added a sentence before the plot about the slower running.

Thanks.